ado_asana_sync/sync/sync.py

```diff
@@ -125,44 +241,72 @@ def sync_project(a: app, project):
             )
             continue
         asana_matched_user = matching_user(asana_users, ado_assigned_email)
         if asana_matched_user == None:
             continue
 
-        current_work_item = work_item(
-            ado_id=ado_task.id,
-            title=ado_task.fields["System.Title"],
-            description=ado_task.fields.get("System.Description"),
-            status=ado_task.fields["System.State"],
-            item_type=ado_task.fields["System.WorkItemType"],
-            created_date=ado_task.fields["System.CreatedDate"],
-            priority=ado_task.fields["Microsoft.VSTS.Common.Priority"],
-            url=ado_task.url,
-            assigned_to=asana_matched_user.gid,
-        )
-
-        # Get the corresponding Asana task by name
-        asana_task = get_asana_task(a, asana_project, current_work_item.asana_title)
-        if asana_task == None:
-            # The Asana task does not exist, create it
-            logging.info(f"creating task {current_work_item.asana_title}")
-            create_asana_task(
-                a,
-                asana_project,
-                current_work_item,
-            )
-        else:
-            # The Asana task exists, update it
-            logging.info(f"updating task {current_work_item.asana_title}")
-            update_asana_task(
-                a,
-                asana_task.gid,
-                current_work_item,
+        # Check if this is an already mapped item.
+        item = TaskItem.find_by_ado_id(a, ado_task.id)
+        if item == None:
+            logging.info(f"{ado_task.fields['System.Title']}:unmapped task")
+            item = TaskItem(
+                ado_id=ado_task.id,
+                ado_rev=ado_task.rev,
+                title=ado_task.fields["System.Title"],
+                item_type=ado_task.fields["System.WorkItemType"],
+                created_date=iso8601_utc(datetime.now()),
+                updated_date=iso8601_utc(datetime.now()),
+                url=ado_task.url,
+                assigned_to=asana_matched_user.gid,
             )
+            # Check if there is a matching asana task with a matching title.
+            asana_task = get_asana_task_by_name(a, asana_project, item.asana_title)
+            if asana_task == None:
+                # The Asana task does not exist, create it and map the tasks
+                logging.info(
+                    f"{ado_task.fields['System.Title']}:no matching asana task exists, creating new task"
+                )
+                create_asana_task(
+                    a,
+                    asana_project,
+                    item,
+                )
+                continue
+            else:
+                # The Asana task exists, map the tasks in the db
+                logging.info(
+                    f"{ado_task.fields['System.Title']}:found a matching asana task by name, updating task"
+                )
+                item.asana_gid = asana_task.gid
```

Comment:
**Suggestion:** Consider adding a check to ensure that the `asana_task` object returned by the `get_asana_task_by_name` function is not `None` before accessing its `gid` attribute. This will prevent a `NoneType` error if the function returns `None`.
```suggestion
                if asana_task is not None:
                    item.asana_gid = asana_task.gid
```